### PR TITLE
6256 Update staff comments

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#6256

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/business/business_comments.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+@cors_preflight('GET, POST')
+@API.route('/<string:identifier>/comments', methods=['GET', 'POST', 'OPTIONS'])
+class BusinessCommentResource(Resource):
+    """Business Comment service."""
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_auth
+    def get(identifier):
```
Comment (lekshmimallika-aot) [inline legal-api/src/legal_api/resources/business/business_comments.py:44]: get by id is something that exists for all other objects. May have to check whether that should be supported

File: legal-api/src/legal_api/resources/business/business_comments.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            comment.comment = json_input['comment']['comment']
+            comment.staff_id = user.id
+            comment.business_id = business.id
+            comment.timestamp = datetime.datetime.utcnow()
+            comment.save()
+        except BusinessException as err:
+            reply = json_input
+            reply['errors'] = [{'error': err.error}, ]
+            return jsonify(reply), err.status_code or \
+                (HTTPStatus.CREATED if (request.method == 'POST') else HTTPStatus.ACCEPTED)
```
Comment (lekshmimallika-aot) [inline legal-api/src/legal_api/resources/business/business_comments.py:98]: If I understand correctly this gets called for a POST . Should this check be there

File: legal-api/src/legal_api/resources/business/business_comments.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        rv = []
+        for comment in comments:
+            rv.append(comment.json)
+
+        return jsonify(comments=rv)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_auth
+    def post(identifier):
```
Comment (lekshmimallika-aot) [inline legal-api/src/legal_api/resources/business/business_comments.py:63]: I have not gone through requirements. Is there a need to edit a comment? In that case PUT should be supported as well

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:44]: id of comments?

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 3 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:63]: no. they cannot edit

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 1 above)
Comment (lekshmimallika-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:44]: ya

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:44]: as per the requirement this is not required. but if we follow that standard i can include it

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 2 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:98]: will fix it

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 1 above)
Comment (lekshmimallika-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:44]: I think get all, get by id is more of a std for an endpoint

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:44]: will include that

**Review by lekshmimallika-aot — APPROVED** [review]

File: legal-api/src/legal_api/resources/business/business_comments.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+        # save comment
+        user = User.get_or_create_user_by_jwt(g.jwt_oidc_token_info)
+        try:
+            comment = Comment()
+            comment.comment = json_input['comment']['comment']
+            comment.staff_id = user.id
+            comment.business_id = business.id
+            comment.timestamp = datetime.datetime.utcnow()
+            comment.save()
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/business/business_comments.py:101]: Does this reuse the existing comments table?

Where is field `comment.submitterDisplayName`?

File: legal-api/src/legal_api/resources/business/business_comments.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    @jwt.requires_auth
+    def get(identifier, comment_id=None):
+        """Return a JSON object with meta information about the Service."""
+        # basic checks
+        business = Business.find_by_identifier(identifier)
+        err_msg, err_code = BusinessCommentResource._basic_checks(identifier, business, request)
+        if err_msg:
+            return jsonify(err_msg), err_code
+
+        comments = db.session.query(Comment).filter(Comment.business_id == business.id, Comment.filing_id.is_(None))
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/business/business_comments.py:53]: If we are reusing the existing comments table, does the result set need to be filtered by a comment type? Or is it enough that the comments we have have a business_id while the other have a filing_id?

File: legal-api/tests/unit/api/test_business_comments.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+del SAMPLE_JSON_DATA['comment']['timestamp']
+del SAMPLE_JSON_DATA['comment']['submitterDisplayName']
+
+
+def test_get_all_business_comments_no_results(session, client, jwt):
+    """Assert that endpoint returns no-results correctly."""
+    identifier = 'CP7654321'
+    factory_business(identifier)
+
+    rv = client.get(f'/api/v1/businesses/{identifier}/comments',
```
Comment (severinbeauvais) [inline legal-api/tests/unit/api/test_business_comments.py:44]: 👍

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 4 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:101]: Yes its existing comments table. `submitterDisplayName` is `display_name` from `user` table, which will be in get response

File: legal-api/src/legal_api/resources/business/business_comments.py (hunk 5 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_comments.py:53]: filing comments api is inserting only filing_id and the existing schema says one of the id required. so this should be enough to separate comments

**Review by severinbeauvais — APPROVED** [review]
Looks good.

**Review by severinbeauvais — APPROVED** [review]
Version updates look good.

**Pull request merged**